packages/template-retail-react-app/app/commerce-api/contexts.js

```diff
@@ -13,6 +19,26 @@ export const CommerceAPIContext = createContext()
 export const CommerceAPIProvider = CommerceAPIContext.Provider
 export const useCommerceAPI = () => useContext(CommerceAPIContext)
 
+/**
+ * Extract the React SDK providers used across the app
+ * This combines QueryClient, Config, CommerceApi, and Auth contexts into one reusable component
+ */
+export const CommerceSDKReactProvider = ({api, children}) => {
+    const queryClient = new QueryClient()
```

Comment:
nit: We can use `useMemo` to avoid creating a new `QueryClient` on every render.

```suggestion
     const queryClient = React.useMemo(() => new QueryClient(), [])
```